src/Normalizer/NormalizerBase.php

```diff
@@ -36,8 +22,8 @@ public function supportsDenormalization($data, $type, $format = NULL) {
   /**
    * {@inheritdoc}
    */
-  public function denormalize($data, $class, $format = NULL, array $context = array()) {
-    return FALSE;
+  public function denormalize($data, $class, $format = NULL, array $context = []) {
+    throw new \RuntimeException('Denormalization is not supported.');
```

Comment:
I could not see the code you are referring to. The class you mention does not implement `denormalize()`, only `supportsDenormalization()`.

The closest I could find is `Drupal\hal\Normalizer\ContentEntityNormalizer` and that one does throw exceptions during errors.

![web 2017-02-12 08-16-46](https://cloud.githubusercontent.com/assets/1140906/22860357/9f44b474-f0fb-11e6-9203-504baceffb5b.png)